### PR TITLE
Refactor Build Pipeline

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR refactors the build pipeline simplifying its configuration and deleting legacy code:

- [x] Fix Doc pipeline bug
- [x] Fix Fork Build

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -25,78 +64,43 @@ jobs:
...
+        uses: docker/build-push-action@v2
         with:
-          name: liqo/${{ matrix.component }}-ci
-          dockerfile: build/${{ matrix.component }}/Dockerfile
-          push: false
-        if: github.ref != 'refs/heads/master' && github.event.pull_request.head.repo.full_name != 'liqotech/liqo' &&
-            !startsWith(github.ref, 'refs/tags/v')
+          tags: |
+            liqo/${{ matrix.component }}:latest
+            liqo/${{ matrix.component }}:${{ needs.configure.outputs.commit_ref }}
```
Comment (giorio94) [inline .github/workflows/integration.yml:89]: Looks like `repo-suffix` is not used.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -106,18 +110,21 @@ jobs:
          run: echo "branch_owner=`git log -1 --format='%ae' $commit_ref^!`"
        - name: "Install python dependencies"
          run: pip install requests
-       - name: "Set branch name"
-         run: echo "branch_name=`echo ${GITHUB_REF#refs/*/}`" >> $GITHUB_ENV
```
Comment (giorio94) [inline .github/workflows/integration.yml:110]: If this part is deleted, then the steps before could be deleted as well

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -213,29 +210,37 @@ jobs:
         tar -zxvf kubebuilder_${version}_linux_${arch}.tar.gz
         mv kubebuilder_${version}_linux_${arch} kubebuilder && sudo mv kubebuilder /usr/local/
         export PATH=$PATH:/usr/local/kubebuilder/bin
```
Comment (giorio94) [inline .github/workflows/integration.yml:212]: This line could be dropped as (if I am not wrong) it is not persisted to next step

File: .github/workflows/integration.yml (hunk 1 above)
Comment (palexster) [reply .github/workflows/integration.yml:89]: Nice catch!

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -25,99 +64,61 @@ jobs:
...
+      - name: Build (and Publish) ${{ matrix.component }} image
+        uses: docker/build-push-action@v2
         with:
-          name: liqo/${{ matrix.component }}-ci
-          dockerfile: build/${{ matrix.component }}/Dockerfile
-          push: false
-        if: github.ref != 'refs/heads/master' && github.event.pull_request.head.repo.full_name != 'liqotech/liqo' &&
-            !startsWith(github.ref, 'refs/tags/v')
+          tags: |
+            liqo/${{ matrix.component }}:latest
```
Comment (giorio94) [inline .github/workflows/integration.yml:88]: Probably the suffix should be added also here

**Review by giorio94 — APPROVED** [review]
/lgtm

Just one minor comment inline

File: .github/workflows/integration.yml [hunk 5]
```diff
@@ -25,99 +64,61 @@ jobs:
         - init-pod-mutator
         - pod-mutator
         - peering-request-webhook-init
-        - crdReplicator
+        - crdreplicator
```
Comment (mlavacca) [inline .github/workflows/integration.yml:67]: can we use the same naming convention? I think that kebab-case fits better than flat-case, just an opinion, though.

**Review by mlavacca — APPROVED** [review]

**Pull request merged**